Documentation/api.md

```diff
@@ -500,6 +500,8 @@ etcd can be used as a centralized coordination service in a cluster, and `Compar
 
 This command will set the value of a key only if the client-provided conditions are equal to the current conditions.
 
+_Note that `CompareAndSwap` does not work with [directories](#listing-a-directory)._
```

Comment:
Can we also mention that a 102 "Not a file" error will be returned? I think that is the most confusing part since we also return this error when deleting a dir without setting dir option.